### PR TITLE
fixes flaky TestListClusters unit test

### LINKED ISSUE #1498: Investigate flaky TestListClusters/scenario_1:_list_clusters_that_belong_to_the_given_project
Failed on first run here: https://drone.loodse.com/kubermatic/kubermatic/2155/10

I restarted the build and it passed:  https://drone.loodse.com/kubermatic/kubermatic/2157/10

### PR DESCRIPTION
**Pull request by p0lyn0mial**
**What this PR does / why we need it**: fixes flaky TestListClusters unit test

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1498 

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: api/pkg/handler/cluster.go [hunk 1]
```diff
@@ -225,7 +225,7 @@ func newListClusters(projectProvider provider.ProjectProvider) endpoint.Endpoint
 			return nil, kubernetesErrorToHTTPError(err)
 		}
 
-		clusters, err := clusterProvider.List(project, nil)
+		clusters, err := clusterProvider.List(project, &provider.ClusterListOptions{SortBy: "metadata.creationTimestamp"})
```
Comment (p0lyn0mial) [inline api/pkg/handler/cluster.go:228]: huh, I forgot to actually pass sorting key.

File: api/pkg/handler/cluster.go [hunk 2]
```diff
@@ -263,11 +263,6 @@ func newDeleteCluster(sshKeyProvider provider.NewSSHKeyProvider, projectProvider
 			return nil, kubernetesErrorToHTTPError(err)
 		}
 
-		err = clusterProvider.Delete(user, project, req.ClusterName)
-		if err != nil {
-			return nil, kubernetesErrorToHTTPError(err)
-		}
-
 		// TODO: I think that in general it would be better if the cluster resource
```
Comment (p0lyn0mial) [inline api/pkg/handler/cluster.go:266]: what do you think about this ?

File: api/pkg/handler/cluster.go [hunk 3]
```diff
@@ -282,6 +277,11 @@ func newDeleteCluster(sshKeyProvider provider.NewSSHKeyProvider, projectProvider
 				return nil, kubernetesErrorToHTTPError(err)
 			}
 		}
+
+		err = clusterProvider.Delete(user, project, req.ClusterName)
```
Comment (p0lyn0mial) [inline api/pkg/handler/cluster.go:281]: just moved the cluster deletion a few lines down, if an error occurs while removing/detaching ssh keys from the cluster the user will repeat the request and hopefully this time everything will works.

File: api/pkg/handler/cluster.go (hunk 2 above)
Comment (mrIncompetent) [reply api/pkg/handler/cluster.go:266]: sgtm

**Review by mrIncompetent — APPROVED** [review]

File: api/pkg/handler/cluster.go (hunk 2 above)
Comment (p0lyn0mial) [reply api/pkg/handler/cluster.go:266]: will add issue for that.

**Pull request merged**